framework/src/base/FEProblemBase.C

```diff
@@ -2815,21 +2815,29 @@ FEProblemBase::execMultiAppTransfers(ExecFlagType type, MultiAppTransfer::DIRECT
   {
     const auto & transfers = wh.getActiveObjects();
 
+    #ifndef NDEBUG
     _console << COLOR_CYAN << "\nStarting Transfers on " <<  Moose::stringify(type) << string_direction << "MultiApps" << COLOR_DEFAULT << std::endl;
+    #endif
     for (const auto & transfer : transfers)
     {
       Moose::perf_log.push(transfer->name(), "Transfers");
       transfer->execute();
       Moose::perf_log.pop(transfer->name(), "Transfers");
     }
 
+    #ifndef NDEBUG
     _console << "Waiting For Transfers To Finish" << '\n';
     MooseUtils::parallelBarrierNotify(_communicator);
```

Comment:
I see what you mean.  Didn't notice the barrier at the end of that function. Good catch.

Putting the `#ifndef` around just the lines that output inside parallelBarrierNotify would accomplish the same thing.